controllers/exam.controller.js

```diff
@@ -0,0 +1,63 @@
+const examService = require("../services/exam.service");
+
+async function createExam(req, res) {
+	try {
+		const examData = {
+			creator: req.session.user.userId,
+			title: req.body.title,
+			description: req.body.description,
+			startDate: req.body.startDate,
+			duration: req.body.duration,
+			rootHash: req.body.rootHash,
+			secretKey: req.body.secretKey,
+			questions: req.body.questions,
+		};
+
+		const { newExam, questions } = await examService.create(
+			examData,
+			examData.questions
+		);
+
+		return res.status(200).json({
+			message: "Exam created successfully",
+			newExam,
+			questions,
+		});
+	} catch (err) {
+		console.error(err);
+		return res.status(500).json({ message: "Internal server error" });
+	}
+}
+
+async function getAllExams(req, res) {
+	try {
+		const exams = await examService.getAllByUser(req.session.user.userId);
+		res.status(200).json(exams);
+	} catch (err) {
+		console.error(err);
+		res.status(500).json({ message: "Internal Server Error" });
+	}
+}
+
+async function getExamById(req, res) {
+	try {
+		const result = await examService.getByIdWithParticipation(
+			req.params.id,
+			// Unnecessary Control?
+			req.session.user ? req.session.user.userId : null
```

Comment:
This is a control to check if user has session. It would throw error on req.session.user.userId so you would have to use req.sessions.user?.userId .  That would shorten that line but necessary

Reply:
Yeah, got it. But We have an authorization check from middleware which is used for these routes already. So, I think we don't need this control.

Reply:
In the past getExamById could return exam details without user session, that might be handy for us for future. It should return only exam if no session, return isFinished if there is session. But you can remove the unnecessary check for getParticipation service function

Reply:
I understood your approach. I will implement it but our older version auth control was there for this route, and it was working interestingly. anyway, I m gonna do it.

Reply:
Oh then there was a weird mistake there that we forgot, ok cool thx good job  😁